rcl/src/rcl/node.c

```diff
@@ -628,13 +628,7 @@ rcl_ret_t rcl_node_type_description_service_fini(rcl_node_t * node)
     return RCL_RET_NOT_INIT;
   }
 
-  const rcl_ret_t ret =
-    rcl_service_fini(&node->impl->get_type_description_service, node);
-  if (RCL_RET_OK == ret) {
-    node->impl->get_type_description_service = rcl_get_zero_initialized_service();
-  }
-
-  return ret;
+  return RCL_RET_OK;
```

Comment:
> This was causing a segmentation fault in the new rclcpp::WaitSet implementation as we were waiting on a service that was being cleaned up by the fini call.

i am not sure about this problem, can you elaborate it? btw, after the change, this function does pretty much nothing? in that case, we can just remove this function?

Reply:
Sorry, I opened this to get @clalancette's opinion quickly.  I have updated the PR as well as the description to more clearly explain the problem and proposed solution.